---
title: notice_error (Python API d'agent)
type: apiDoc
shortDescription: Enregistre les détails d'une exception Python en tant qu'erreur.
tags:
  - Agents
  - Python agent
  - Python agent API
metaDescription: 'Python API: This call records the details of a Python exception as an error.'
freshnessValidatedDate: never
translationType: machine
---

## Syntaxe

```py
newrelic.agent.notice_error(error=None, attributes={}, expected=None, ignore=None, status_code=None, application=None)
```

Enregistre les détails d&apos;une exception Python en tant qu&apos;erreur.

## Description

Par défaut, l&apos;agent Python signale uniquement les exceptions non gérées. Utilisez `notice_error` pour enregistrer toute exception Python comme une erreur, qui peut ensuite être trouvée dans l&apos; UI. Si aucun paramètre n&apos;est fourni, les détails de l&apos;exception actuellement traitée seront utilisés.

Vous pouvez enregistrer jusqu&apos;à cinq exceptions distinctes par [transaction](/docs/accounts-partnerships/education/getting-started-new-relic/glossary#transaction) et jusqu&apos;à 20 exceptions au total pour toutes les transactions par [cycle de collecte](/docs/apm/new-relic-apm/getting-started/glossary#harvest-cycle).

Lorsque `notice_error()` est appelé dans le contexte d&apos;une demande Web de monitoring ou d&apos;une tâche en arrière-plan, les détails de l&apos;exception seront signalés par rapport à l&apos; application à laquelle la demande ou la tâche est signalée.

Si appelé en dehors du contexte d&apos;une requête Web de monitoring ou d&apos;une tâche d&apos;arrière-plan, l&apos;appel sera ignoré à moins que l&apos;argument de mot-clé [`application`](/docs/agents/python-agent/python-agent-api/application) ne soit fourni et qu&apos;un objet application correspondant à l&apos; application par rapport à laquelle l&apos;exception doit être enregistrée soit fourni. Un objet d&apos;application approprié peut être obtenu à l&apos;aide de la fonction `newrelic.agent.application()` .

## Paramètres

<Callout variant="tip">
  Dans presque tous les cas, `notice_error` ne nécessitera aucun paramètre.
</Callout>

<table>
  <thead>
    <tr>
      <th width="25%">
        paramètres
      </th>

      <th>
        Description
      </th>
    </tr>
  </thead>

  <tbody>
    <tr>
      <td>
        `error`

        *uplet*
      </td>

      <td>
        Facultatif et rarement utilisé. Un uplet contenant des informations d&apos;exception `(exception_class, exception_instance, traceback)` renvoyées par [`sys.exc_info()`](https://docs.python.org/2/library/sys.html#sys.exc_info).
      </td>
    </tr>

    <tr>
      <td>
        `attributes`

        *dict*
      </td>

      <td>
        Facultatif. attribut personnalisé à ajouter à l&apos;événement d&apos;erreur (en plus de tout attribut personnalisé déjà ajouté à la transaction). Si [le mode haute sécurité](/docs/agents/manage-apm-agents/configuration/high-security-mode) est activé, cela ne fonctionnera pas.
      </td>
    </tr>

    <tr>
      <td>
        `expected`

        *boolean*, *iterable\[String]*, *callable(exception\_class, exception\_instance, traceback)-&gt;boolean*
      </td>

      <td>
        Facultatif. Les erreurs à marquer comme attendues peuvent être transmises sous la forme d&apos;un itérable de chaînes sous la forme `module:class`. Cette valeur peut également être un appelable ou un booléen indiquant si l&apos;erreur est attendue. Ces erreurs seront signalées à l&apos; UI mais n&apos;affecteront pas le score Apdex ou le taux d&apos;erreur.
      </td>
    </tr>

    <tr>
      <td>
        `ignore`

        *boolean*, *iterable\[String]*, *callable(exception\_class*, *exception\_instance, traceback)-&gt;boolean*
      </td>

      <td>
        Facultatif. Les erreurs à ignorer peuvent être transmises sous la forme d&apos;un itérable de chaînes sous la forme `module:class`. Cette valeur peut également être un appelable ou un booléen indiquant si l&apos;erreur doit être ignorée. Utile lorsque certains types d’exceptions doivent toujours être ignorés et jamais enregistrés.
      </td>
    </tr>

    <tr>
      <td>
        `status_code`

        *chaîne*, *entier*, *appelable(exception\_class, exception\_instance, traceback)*
      </td>

      <td>
        Facultatif. Le code d&apos;état d&apos;exception. Cette valeur peut être une chaîne, un entier ou un appelable qui prend les informations d&apos;exception `(exception_class, exception_instance, traceback)` renvoyées par [`sys.exc_info()`](https://docs.python.org/2/library/sys.html#sys.exc_info) et renvoie le code d&apos;état sous forme d&apos;entier.
      </td>
    </tr>

    <tr>
      <td>
        `application`

        *objet d&apos;application*
      </td>

      <td>
        Facultatif. Si appelé en dehors du contexte d&apos;une requête Web de monitoring ou d&apos;une tâche d&apos;arrière-plan, l&apos;appel sera ignoré à moins que l&apos; [objet`application` ](/docs/agents/python-agent/python-agent-api/application)ne soit fourni.
      </td>
    </tr>
  </tbody>
</table>

## Valeurs de retour

Aucun.

## Exemples

### Exemple simple de signalement d&apos;exceptions [#simple-example]

Dans la grande majorité des cas, vous n&apos;aurez pas besoin de passer de paramètre. Vous devez simplement appeler ce qui suit lorsque vous souhaitez signaler une exception :

```py
newrelic.agent.notice_error()
```

### Exemple utilisant un booléen [#boolean-example]

Un exemple de `notice_error` utilisant une valeur booléenne. Cela indique qu’une erreur est à prévoir.

```py
newrelic.agent.notice_error(expected=True)
```

### Appel avec l&apos;uplet sys.exc\_info() et un paramètre supplémentaire [#complex-example]

Un exemple de `notice_error` utilisant les données `sys.exc_info()` :

```py
def complex_ignore_errors(exc, val, tb):
   # do some logic here
   return False

newrelic.agent.notice_error(attributes={'my_special_exception': True}, ignore=complex_ignore_errors)
```

### Exemple d&apos;utilisation du rappel [#callback-example]

Si vous devez filtrer les exceptions de manière dynamique en fonction de l&apos;attribut d&apos;un type d&apos;exception spécifique, vous pouvez fournir une fonction de rappel :

```py
def _ignore_errors(exc, value, tb):
    if instance(value, HTTPError):
        if value.status == 404:
            return True

newrelic.agent.notice_error(ignore=_ignore_errors)
```

Si l&apos;exception doit être ignorée/attendue, définissez la valeur de retour de l&apos;appelable sur `True`. Renvoie `False` si l&apos;exception ne doit jamais être ignorée/attendue, quelles que soient les autres vérifications, et `None` si les vérifications ultérieures et les règles intégrées doivent déterminer si l&apos;exception doit être ignorée/attendue. Un rappel renverrait normalement soit `True` soit `None`.